Java/core/src/main/java/com/amazon/randomcutforest/state/sampler/CompactSamplerMapper.java

```diff
@@ -78,6 +79,9 @@ public CompactSamplerState toState(CompactSampler model) {
         state.setSize(model.size());
         state.setCapacity(model.getCapacity());
         state.setLambda(model.getLambda());
+        state.setLastUpdateOfLambda(model.getLastUpdateOflambda());
+        state.setAccumulatedLambda(model.getAccumulatedLambda());
+        state.setMostRecentTimeStamp(model.getMostRecentTimeStamp());
```

Comment:
discussion. For some clients, this feature might not be actually used and the fields could be unnecessary overhead from unused features. We might be able to design a way out for them. Or, such feature/enhancement-specific data will bloat the core model.

Reply:
True, but the main point is that a versatile model is more useful. What is the core model of a sketch of data :) This is three numbers per sampler -- say 150 numbers overall and the gain is that the samplers can perform dynamically adjusted sampling. For time series application, that's a plus. One way to design would be to create version flags and serialization based on those version flags -- but that would also be a trap for users to get stuck with serialized models.

Reply:
For example, we are only using samplers with the same parameters -- but there is no reason to do so! Maybe some trees focus on weekly variations and some focus on daily variations. 
Consider the version 1.0 ser/De where the timestamps were present (an option in the current version) -- the fact that those timestamps were present meant one could perform an insertion based on time stamps in deserialization. Sure, in this revamp when the trees can be serialized as well that information (of timestamps) is less critical. But that was handy in 1.0 -- because they were available.

Reply:
for design, serde could rely on some config to control which data to preserve. by default, the config should preserve the core model, which should also be the default forest config.

for example,
rcf = new rcf() // default
rcf_state = new mapper().map(rcf) // default data

rcf = new rcf().enableX().enableY()
rcf_state_with_x_y = new mapper().map(rcf) // state w/ because rcf config has x and y enabled
rcf_state_w/0_x_y = new mapper().disableX().disableY().map(rcf) // state w/o x and y because mapper has x and y explicitly disabled

The goal is to protect the model state from having to take on any data from any possibly used feature, many of which may just be some default values from never being used.

For this one, there may be enableDynamicLambda config at forest level. The default could be false, as the current behavior. If a client explicitly enables this, this operation will be allowed and the relevant data, used or not, will be mapped to state by (mapper) default. Otherwise, the state fields will be null-valued that serializer can skip.

Reply:
Makes sense, I'm open to it. But I thought the issue was that the fields had to be always present (say for protostuff, which needed a schema - but I can be wrong). And if the field is present, then the issue is null vs default value. But yes, these can add up. Josh was looking into making the mapper efficient (copy of arrays) so maybe we have this for consideration there?

Reply:
I'm open to adding configuration options to mappers (like `saveTreeState` and `saveExecutorContext`) but I'm not sure if the savings in this case is worth the additional logic. For 50 trees and sample size 256 we're currently getting JSON files around 1MB. In the latest revision, we're adding 2 long fields per sampler, so 800 bytes in the 50 tree case. If we want a mapper to be able to output a state class without those long fields, we would need two sampler state classes (one which includes them and one which doesn't), and we'd have to thread that logic through the RandomCutForestMapper, which is already a bit tangled from dealing with compact enabled/disabled, precision, optional tree state, etc.

Reply:
since lambda is a public parameter core to rcf and there is a getter, a matching setter and its overheads are justifiable.